kratos/processes/from_json_check_result_process.h

```diff
@@ -277,51 +277,54 @@ class KRATOS_API(KRATOS_CORE) FromJSONCheckResultProcess
         for (auto& p_var_double : mpNodalVariableDoubleList) {
             const auto& r_var_database = r_node_database.GetVariableData(*p_var_double);
 
-            check_counter = IndexPartition<std::size_t>(r_nodes_array.size()).for_each<SumReduction<IndexType>>([&] (std::size_t Index){
+            check_counter = IndexPartition<std::size_t>(r_nodes_array.size()).for_each<SumReduction<IndexType>>([&it_node_begin, &p_var_double, &r_var_database, &time, this] (std::size_t Index){
+                IndexType r_check_counter = 0;
                 auto it_node = it_node_begin + Index;
-                const double result = GetValue<THistorical>(it_node, p_var_double);
+                const double result = this->GetValue<THistorical>(it_node, p_var_double);
                 const double reference = r_var_database.GetValue(Index, time);
                 if (!CheckValues(result, reference)) {
                     FailMessage(it_node->Id(), "Node", result, reference, p_var_double->Name());
-                    check_counter += 1;
+                    r_check_counter += 1;
                 }
-                return check_counter;
+                return r_check_counter;
             });
         }
 
         for (auto& p_var_array : mpNodalVariableArrayList) {
             const auto& r_var_database = r_node_database.GetVariableData(*p_var_array);
 
-            check_counter = IndexPartition<std::size_t>(r_nodes_array.size()).for_each<SumReduction<IndexType>>([&] (std::size_t Index){
+            check_counter = IndexPartition<std::size_t>(r_nodes_array.size()).for_each<SumReduction<IndexType>>([&it_node_begin, &p_var_array, &r_var_database, &time, this] (std::size_t Index){
+                IndexType r_check_counter = 0;
```

Comment:
```suggestion
                IndexType counter = 0;
```

the notation `r_` is reserved for references

Reply:
Done